aerogear/app-metrics-operator
# Provision everything on CR creation

JIRA: https://issues.jboss.org/browse/AEROGEAR-9187

This is mostly just copypasta from the unifiedpush-operator, since from a deployment point-of-view they're very similar.

----

# Verification steps

* Checkout this branch
* As a cluster admin user in OpenShift (because of role creation), run `make cluster/prepare`. This will create an app-metrics project and other things needed for the operator.
* Deploy the operator using the image built for this PR:
```
cat deploy/operator.yaml | sed -e "s|quay.io/aerogear/app-metrics-operator:master|quay.io/aerogear/app-metrics-operator:$(git rev-parse --short HEAD|g" | kubectl create -f -
```
* Check that the operator has started successfully with like `kubectl get po` or whatever, and have a look at the logs to make sure there's no obvious errors in there.

* Create an app-metrics instance by adding a CR! :smiley_cat: :
```
kubectl apply -f deploy/crds/metrics_v1alpha1_appmetricsservice_cr.yaml
```

* Once everything has been created, you should be able to login to the service, through the oauth-proxy by going to the route in your browser. On linux, something like this should probably work to open it in your default browser:
```
xdg-open https://$(oc get route -o json | jq -r .items[0].spec.host)
```

* If you're happy with it, you can optionally remove everything again by running `make cluster/clean`.
